src/test/components/ProfileViewerContainer.test.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import { shallowWithStore } from '../fixtures/enzyme';
+
+import ProfileViewerContainer from '../../components/app/ProfileViewerContainer';
+import { changeSelectedTab } from '../../actions/app';
+import { storeWithProfile } from '../fixtures/stores';
+import { getProfileFromTextSamples } from '../fixtures/profiles/make-profile';
+
+import type { TabSlug } from '../../types/actions';
+
+describe('app/ProfileViewerContainer', function() {
+  const { profile } = getProfileFromTextSamples(`
+    A A A
+    B B B
+    C C H
+    D F I
+    E E
+  `);
+
+  const tabSlugs = [
+    'stack-chart',
+    'marker-chart',
+    'flame-graph',
+    'marker-table',
+    'calltree', // 'calltree' not first on purpose
+  ];
+
+  it('renders an initial view', () => {
+    const store = storeWithProfile(profile);
+    // dive() will shallow-render the wrapped component
+    const view = shallowWithStore(<ProfileViewerContainer />, store);
+    expect(view.dive()).toMatchSnapshot();
+  });
+
+  tabSlugs.forEach((tabSlug: TabSlug) => {
+    it(`renders initial views with tab ${tabSlug}`, () => {
+      const store = storeWithProfile(profile);
+      store.dispatch(changeSelectedTab(tabSlug));
+
+      const view = shallowWithStore(<ProfileViewerContainer />, store);
+      expect(view.dive()).toMatchSnapshot();
+    });
+  });
+
+  it('can move between tabs', () => {
+    const store = storeWithProfile(profile);
+    const view = shallowWithStore(<ProfileViewerContainer />, store);
+
+    tabSlugs.forEach(tabSlug => {
+      store.dispatch(changeSelectedTab(tabSlug));
+      view.update();
+      expect(view.dive()).toMatchSnapshot();
```

Comment:
we call `dive` inside the `expect` call, because we need to `update` the connected component, not the wrapped component, so we need to keep `view` pointing to the connected component.

Reply:
This would be good documentation for how to use `shallowWithStore` in `fixtures/enzyme.js`.

Reply:
Yep you're right, I'll do